File: frontend/src/types/shipment.ts
// ============================================================
// Shipment Types — Kargo / gönderi veri yapısı tanımları
// ============================================================
// Bu dosya, kargo gönderilerinin veri şeklini tanımlar.
// "ShipmentStatus" tipi, bir gönderinin alabileceği durumları sınırlar.
// "Shipment" arayüzü, takip kodu, kargo firması, gecikme bilgisi gibi
// tüm gönderi detaylarını taşır.
// ============================================================

/** Bir gönderinin alabileceği tüm durumlar */
export type ShipmentStatus =
  | "Yolda"           // Kargo yolda, taşınıyor
  | "Teslim Edildi"   // Alıcıya başarıyla ulaştı
  | "Gecikti"         // Tahmini tarihten sonra hâlâ teslim edilmedi
  | "İade"            // Alıcıdan gönderene iade sürecinde
  | "Beklemede";      // Henüz kargoya verilmedi

/** Tek bir kargo gönderisini temsil eden arayüz */
export interface Shipment {
  id: string;              // Benzersiz gönderi kodu, örn. "SHP-1001"
  orderId: string;         // Bağlı sipariş numarası, örn. "ORD-1001"
  carrier: string;         // Kargo firması adı, örn. "Yurtiçi Kargo"
  trackingCode: string;    // Kargo takip numarası
  status: ShipmentStatus;  // Gönderinin mevcut durumu
  origin: string;          // Çıkış şehri
  destination: string;     // Varış şehri
  estimatedDelivery: string; // Tahmini teslimat tarihi (ISO formatı)
  isDelayed: boolean;      // Gecikme var mı? (DelayFlag bileşeni için)
  delayReason?: string;    // Gecikme nedeni (opsiyonel)
}

File: backend/app/models/shipment.py
from sqlalchemy import Column, String, Boolean
from app.core.database import Base

class Shipment(Base):
    __tablename__ = "shipments"

    id = Column(String, primary_key=True, index=True) # e.g. "SHP-1001"
    orderId = Column(String, nullable=False)
    carrier = Column(String, nullable=False)
    trackingCode = Column(String, nullable=False)
    status = Column(String, nullable=False)
    origin = Column(String, nullable=False)
    destination = Column(String, nullable=False)
    estimatedDelivery = Column(String, nullable=False)
    isDelayed = Column(Boolean, default=False)
    delayReason = Column(String, nullable=True)
